deal/core.py

```diff
@@ -198,3 +206,84 @@ def __call__(self, _class):
 
         # return update_wrapper(patched_class, _class)
         return patched_class
+
+
+class Raises(_Base):
+    exception = exceptions.RaisesContractError
+
+    def __init__(self, *exceptions, message=None, exception=None, debug=False):
+        """
+        Step 1. Set allowed exceptions list.
+        """
+        self.exceptions = exceptions
+        super(Raises, self).__init__(
+            validator=None,
+            message=message,
+            exception=exception,
+            debug=debug,
+        )
+
+    def patched_function(self, *args, **kwargs):
+        """
+        Step 3. Wrapped function calling.
+        """
+        try:
+            return self.function(*args, **kwargs)
+        except exceptions.ContractError:
+            raise
+        except Exception as exc:
+            if not isinstance(exc, self.exceptions):
+                raise self.exception from exc
+            raise
+
+
+class Offline(_Base):
+    exception = exceptions.OfflineContractError
+
+    def __init__(self, message=None, exception=None, debug=False):
+        """
+        Step 1. Init params.
+        """
+        super(Offline, self).__init__(
+            validator=None,
+            message=message,
+            exception=exception,
+            debug=debug,
+        )
+
+    def patched_function(self, *args, **kwargs):
+        """
+        Step 3. Wrapped function calling.
+        """
+        true_socket = socket.socket
+        socket.socket = self.fake_socket
+        try:
+            return self.function(*args, **kwargs)
+        finally:
+            socket.socket = true_socket
+
+    def fake_socket(self, *args, **kwargs):
+        raise self.exception
+
+
+class PatchedStringIO(StringIO):
+    def __init__(self, exception):
+        self.exception = exception
+
+    def write(self, *args, **kwargs):
+        raise self.exception
+
+
+class Silent(Offline):
+    exception = exceptions.SilentContractError
+
+    def patched_function(self, *args, **kwargs):
+        """
+        Step 3. Wrapped function calling.
+        """
+        true_stdout = sys.stdout
```

Comment:
What about `sys.stderr` and `open()`?

Reply:
I thought we can get in trouble with printing tracebacks. However, now I understand that traceback for exception in function will be printed after out from function. Let's implement it, thank you.

Reply:
stderr fixed, `open()` moved into #8